nsv13/code/modules/overmap/knpc.dm

```diff
@@ -353,9 +382,10 @@ This is to account for sec Ju-Jitsuing boarding commandos.
 	if(!..())
 		return 0
 	var/mob/living/carbon/human/H = HA.parent
-	var/obj/item/gun/G = H.get_active_held_item()
+	var/obj/A = H.get_active_held_item()
+	var/obj/B = H.get_inactive_held_item() //check your other hand, just in case.
 	//We already have a gun
-	if(G && istype(G))
+	if((A && istype(A, /obj/item/gun)) || (B && istype(B, /obj/item/gun)))
 		return 0
 	var/obj/item/gun/G_New = locate(/obj/item/gun) in oview(HA.view_range, H)
 	if(G_New && gun_suitable(H, G_New))
```

Comment:
On a side note despite you not actively editing this line, this should probably iterate through all guns it finds in the range instead of just locate()ing one out of that list, since it will just return exactly one gun and judge based on that single gun, even if there are five suitable ones nearby.

Reply:
Yeah so far there's been a lot more code I've *wanted* to refactor but not had the time for 😅